tests/utils.go

```diff
@@ -245,27 +262,40 @@ func findConditionByType(conditionType cdiv1.DataVolumeConditionType, conditions
 }
 
 // IsPrometheusAvailable decides whether or not we will run prometheus alert/metric tests
-// Currently only running on OpenShift as it comes preconfigured out of the box
-// Later extend this to use the prometheus CRD detection, similar to how the operator decides this
-func IsPrometheusAvailable(client kubernetes.Interface) bool {
-	return utils.IsOpenshift(client)
+func IsPrometheusAvailable(client *extclientset.Clientset) bool {
+	_, err := client.ApiextensionsV1().CustomResourceDefinitions().Get(context.TODO(), "prometheuses.monitoring.coreos.com", metav1.GetOptions{})
+	if k8serrors.IsNotFound(err) {
+		return false
+	}
+	return true
 }
 
 // MakePrometheusHTTPRequest makes a request to the prometheus api and returns the response
 func MakePrometheusHTTPRequest(f *framework.Framework, endpoint string) *http.Response {
-	var token, host string
-	var err error
+	var token, url, monitoringNs string
 	var resp *http.Response
 
+	url = getPrometheusURL(f)
+	monitoringNs = getMonitoringNs(f)
 	gomega.Eventually(func() bool {
-		host, err = RunOcCommand(f, "-n", "openshift-monitoring", "get", "route", "prometheus-k8s", "--template", "{{.spec.host}}")
+		var secretName string
+		sa, err := f.K8sClient.CoreV1().ServiceAccounts(monitoringNs).Get(context.TODO(), "prometheus-k8s", metav1.GetOptions{})
 		if err != nil {
 			return false
 		}
-		token, err = RunOcCommand(f, "-n", "openshift-monitoring", "sa", "get-token", "prometheus-k8s")
+		for _, secret := range sa.Secrets {
+			if strings.HasPrefix(secret.Name, "prometheus-k8s-token") {
+				secretName = secret.Name
+			}
+		}
+		secret, err := f.K8sClient.CoreV1().Secrets(monitoringNs).Get(context.TODO(), secretName, metav1.GetOptions{})
 		if err != nil {
 			return false
 		}
+		if _, ok := secret.Data["token"]; !ok {
+			return false
+		}
+		token = string(secret.Data["token"])
```

Comment:
Am I missing where token is used?

Reply:
`req.Header.Add("Authorization", "Bearer "+token)`

Reply:
Gotcha, its below in the part that was hidden in the review view.